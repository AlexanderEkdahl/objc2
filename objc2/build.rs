fn main() {
    // The script doesn't depend on our code
    println!("cargo:rerun-if-changed=build.rs");

    #[cfg(feature = "exception")]
    {
        use std::env;

        if std::env::var("DOCS_RS").is_ok() {
            // docs.rs doesn't have clang, so skip building this. The
            // documentation will still work since it doesn't need to link.
            return;
        }
        println!("cargo:rerun-if-changed=extern/exception.m");

        let mut builder = cc::Build::new();
        builder.file("extern/exception.m");

        for flag in env::var("DEP_OBJC_0_2_CC_ARGS").unwrap().split(' ') {
            builder.flag(flag);
        }

        builder.compile("librust_objc_try_catch_exception.a");
    }
}
