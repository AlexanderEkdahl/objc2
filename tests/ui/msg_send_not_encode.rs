//! Test that return types that are not `Encode` are not accepted.
use objc2::{class, msg_send};

fn main() {
    unsafe {
        let cls = class!(NSData);
        let _: Vec<u8> = msg_send![cls, new];
    }
}
