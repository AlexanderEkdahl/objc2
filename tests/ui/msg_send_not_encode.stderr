error[E0277]: the trait bound `Vec<u8>: Encode` is not satisfied
   --> ui/msg_send_not_encode.rs:7:26
    |
7   |         let _: Vec<u8> = msg_send![cls, new];
    |                          ^^^^^^^^^^^^^^^^^^^ the trait `Encode` is not implemented for `Vec<u8>`
    |
    = help: the following other types implement trait `Encode`:
              &'a T
              &'a mut T
              ()
              *const T
              *const c_void
              *mut T
              *mut c_void
              AtomicBool
            and 154 others
note: required by a bound in `send_message`
   --> $WORKSPACE/objc2/src/message/mod.rs
    |
    |         R: Encode,
    |            ^^^^^^ required by this bound in `send_message`
    = note: this error originates in the macro `msg_send` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Vec<u8>: Encode` is not satisfied
   --> ui/msg_send_not_encode.rs:10:21
    |
10  |         let _: () = msg_send![cls, newWith: x];
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Encode` is not implemented for `Vec<u8>`
    |
    = help: the following other types implement trait `Encode`:
              &'a T
              &'a mut T
              ()
              *const T
              *const c_void
              *mut T
              *mut c_void
              AtomicBool
            and 154 others
    = note: required because of the requirements on the impl of `MessageArguments` for `(Vec<u8>,)`
note: required by a bound in `send_message`
   --> $WORKSPACE/objc2/src/message/mod.rs
    |
    |         A: MessageArguments,
    |            ^^^^^^^^^^^^^^^^ required by this bound in `send_message`
    = note: this error originates in the macro `msg_send` (in Nightly builds, run with -Z macro-backtrace for more info)
